### PR TITLE
abci: remove setOption 

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

remove Response/Request SetOption from ABCI. 

I couldn't find it being used anywhere.

Closes: #XXX

### CODE REVIEW

**Review by melekes — APPROVED** [review]

**Comment by melekes** [comment]
Is there a PR which updates the spec?

File: proto/tendermint/abci/types.proto [hunk 1]
```diff
@@ -24,7 +24,6 @@ message Request {
     RequestEcho               echo                 = 1;
     RequestFlush              flush                = 2;
     RequestInfo               info                 = 3;
-    RequestSetOption          set_option           = 4;
     RequestInitChain          init_chain           = 5;
```
Comment (marbar3778) [inline proto/tendermint/abci/types.proto:27]: @melekes do we want to fix the number order here?

Comment (melekes) [reply proto/tendermint/abci/types.proto:27]: v0.34 is breaking anyway, so I'd say yes

File: abci/tests/test_app/app.go [hunk 2]
```diff
@@ -48,24 +48,6 @@ func deliverTx(client abcicli.Client, txBytes []byte, codeExp uint32, dataExp []
 	}
 }
 
-/*func checkTx(client abcicli.Client, txBytes []byte, codeExp uint32, dataExp []byte) {
```
Comment (marbar3778) [inline abci/tests/test_app/app.go:58]: dam, whats your take on the possibility of removing counter app and just supporting kvstore? It seems counter does not provide anything extra?

Comment (melekes) [reply abci/tests/test_app/app.go:58]: fine by me

**Pull request merged**